dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationHelper.java

```diff
@@ -0,0 +1,203 @@
+        }
+        final Map<String, Param> params = form.getParams();
+        if(!UtilMethods.isSet(params)){
+            throw new DotDataException("Required Params aren't set.");
+        }
+        final ServiceDescriptor serviceDescriptor = optionalServiceDescriptor.get();
+        validateIncomingParamNames(params.keySet(),serviceDescriptor);
+
+        final Optional<ServiceSecrets> serviceSecretsOptional = serviceIntegrationAPI
+                .getSecretsForService(serviceKey, host, user);
+        if (!serviceSecretsOptional.isPresent()) {
+            //Create a brand new secret for the present service
+            final ServiceSecrets.Builder builder = new ServiceSecrets.Builder();
+            builder.withServiceKey(serviceKey);
+            for (final Entry<String, Param> stringParamEntry : params.entrySet()) {
+                final String name = stringParamEntry.getKey();
+                final Param param = stringParamEntry.getValue();
+                final Secret secret = Secret.newSecret(param.getValue().toCharArray(), param.getType(), param.isHidden());
+                builder.withSecret(name, secret);
+            }
+            serviceIntegrationAPI.saveSecrets(builder.build(), host, user);
+        } else {
+           //Update individual secrets/properties.
+            for (final Entry<String, Param> stringParamEntry : params.entrySet()) {
+                final String name = stringParamEntry.getKey();
+                final Param param = stringParamEntry.getValue();
+                final Secret secret = Secret.newSecret(param.getValue().toCharArray(), param.getType(), param.isHidden());
+                serviceIntegrationAPI.saveSecret(serviceKey, new Tuple2<>(name, secret), host, user);
+            }
+        }
+    }
+
+    private void validateIncomingParamNames(final Set<String> paramNames, final ServiceDescriptor serviceDescriptor)
+            throws DotDataException {
+        final Map<String, Param> serviceDescriptorParams = serviceDescriptor.getParams();
+        if(serviceDescriptorParams.containsKey("*")){
+            //If an aster has been specified we can have whatever param name we want added.
+           return;
+        }
+        for (final String paramName : paramNames) {
+            if(!serviceDescriptor.getParams().containsKey(paramName)){
+                throw new DotDataException(String.format("Params named `%s` can not be matched against service descriptor.",paramName));
+            }
+        }
+    }
+
+    void createServiceIntegration(String serviceKey, final FormDataMultiPart multipart,
+            final User user)
+            throws IOException, DotSecurityException, DotDataException, JSONException {
+
+        for (final BodyPart bodyPart : multipart.getBodyParts()) {
+            final FileInputStream fileInputStream = (FileInputStream)bodyPart.getEntity();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this FileInputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)